core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -307,22 +297,26 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
    */
   private class Watchers(val key: Any) {
 
-    private[this] val operations = new ConcurrentLinkedQueue[T]()
-    private[this] val approximateWatched = new AtomicInteger(0)
+    private case class WatchedOperation(op: T, purged: AtomicBoolean = new AtomicBoolean(false))
```

Comment:
This should probably be moved to a companion object. Otherwise, it will have a reference to `Watchers`, which doesn't seem necessary.